search_handler.py

```diff
@@ -13,24 +13,26 @@ def load(cls):
 
 config = Configuration.load()
 
-
 from core.opensearch import OpenSearchDocument
 from core.util.opds_writer import AtomFeed, OPDSFeed
 from core.external_search import ExternalSearchIndex
 
 def search_handler(event, context):
     search_url = ""
-    input = event.get('input')
-    if input:
-        request_headers = input.get('Headers')
+    if event:
+        request_headers = event.get('headers')
         if request_headers:
             protocol = request_headers.get('X-Forwarded-Proto', '')
             host = request_headers.get('Host', '')
-            path = input.get('path', '')
-            search_url = protocol + '://' + host + path
+            path = event.get('path', '')
+            request_context = event.get("requestContext")
+            if request_context:
+                stage = request_context.get("stage")
```

Comment:
I want to go back and change this in s3/api gateway later so the stage isn't part of the url.